app/domain/authentication/authn_azure/verify_and_decode_azure_token.rb

```diff
@@ -0,0 +1,88 @@
+require 'jwt'
+
+module Authentication
+  module AuthnAzure
+
+    Err = Errors::Authentication::AuthnAzure
+    Log = LogMessages::Authentication::AuthnAzure
+
+    # We use the OIDC log messages for Identity Provider related code
+    OIDCLog = LogMessages::Authentication::AuthnAzure
+
+    VerifyAndDecodeAzureToken = CommandClass.new(
+      dependencies: {
+        # We have a ConcurrencyLimitedCache which wraps a RateLimitedCache which wraps a FetchProviderCertificate class
+        fetch_provider_certificate: ::Util::ConcurrencyLimitedCache.new(
+          ::Util::RateLimitedCache.new(
+            ::Authentication::AuthnOidc::FetchProviderCertificate.new,
+            refreshes_per_interval: 10,
+            rate_limit_interval:    300, # 300 seconds (every 5 mins)
+            logger: Rails.logger
+          ),
+          max_concurrent_requests: 3,
+          logger: Rails.logger
+        ),
+        logger:                     Rails.logger
+      },
+      inputs:       %i(provider_uri token_jwt)
+    ) do
+
+      def call
+        fetch_certs
+        verify_and_decode_azure_token
+      end
+
+      private
+
+      def fetch_certs(force_read: false)
+        @certs = @fetch_provider_certificate.(
+          provider_uri: @provider_uri,
+          refresh: force_read
+        )
+        @logger.debug(OIDCLog::IdentityProviderCertificateFetchedFromCache.new)
+        @certs
+      end
+
+      def verify_and_decode_azure_token
+        # We ensure that the certs are fresh just before we decode the token
```

Comment:
why is the comment needed in here?
the first method called ensure_certs_are_fresh. the second called decoded_token... seems pretty straight forward